test/widgets/profile_test.dart

```diff
@@ -0,0 +1,319 @@
+      );
+
+      final textFinder = find.text('(unknown user)');
+      check(textFinder.evaluate()).length.equals(1);
+    });
+
+    testWidgets('page builds; dm links to correct narrow', (WidgetTester tester) async {
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+      await setupPage(tester,
+        users: [eg.user(userId: 1)],
+        pageUserId: 1,
+        navigatorObserver: testNavObserver,
+      );
+
+      final targetWidget = find.byIcon(Icons.email);
+      await tester.ensureVisible(targetWidget);
+      await tester.tap(targetWidget);
+      check(pushedRoutes).last.isA<WidgetRoute>().page
+        .isA<MessageListPage>()
+        .narrow.equals(DmNarrow.withUser(1, selfUserId: eg.selfUser.userId));
+    });
+
+    testWidgets('page builds; user links render multiple avatars', (WidgetTester tester) async {
+      final users = [
+        eg.user(userId: 1, profileData: {
+          0: ProfileFieldUserData(value: '[2,3]'),
+        }),
+        eg.user(userId: 2, fullName: 'test user2'),
+        eg.user(userId: 3, fullName: 'test user3'),
+      ];
+
+      await setupPage(tester,
+        users: users,
+        pageUserId: 1,
+        customProfileFields: [mkCustomProfileField(0, CustomProfileFieldType.user)],
+      );
+
+      final avatars = tester.widgetList<Avatar>(find.byType(Avatar));
+      check(avatars.map((w) => w.userId).toList())
+        .deepEquals([1, 2, 3]);
+    });
+
+    testWidgets('page builds; ensure long name does not overflow', (WidgetTester tester) async {
+      final longString = 'X' * 400;
+      final user = eg.user(userId: 1, fullName: longString);
+      await setupPage(tester, users: [user], pageUserId: user.userId);
+      check(find.text(longString).evaluate()).isNotEmpty();
+    });
+
+    testWidgets('page builds; ensure long customProfileFields do not overflow', (WidgetTester tester) async {
+      final longString = 'X' * 400;
+      final user = eg.user(userId: 1, fullName: 'fullName', profileData: {
+        0: ProfileFieldUserData(value: longString),
+        1: ProfileFieldUserData(value: longString),
+        2: ProfileFieldUserData(value: 'x'),
+        3: ProfileFieldUserData(value: 'http://example/$longString'),
+        4: ProfileFieldUserData(value: '[2]'),
+        5: ProfileFieldUserData(value: longString),
+        6: ProfileFieldUserData(value: longString),
+      });
+      final user2 = eg.user(userId: 2, fullName: longString);
+
+      await setupPage(tester, users: [user, user2], pageUserId: user.userId,
+        customProfileFields: [
+          mkCustomProfileField(0, CustomProfileFieldType.shortText),
+          mkCustomProfileField(1, CustomProfileFieldType.longText),
+          mkCustomProfileField(2, CustomProfileFieldType.choice,
+            fieldData: '{"x": {"text": "${longString}", "order": "1"}}'),
```

Comment:
Weird, android studio doesn't flag this in the project analysis but running `flutter analyze` on the cli does flag it. I'll make sure to get better at checking these before pushing

Reply:
Hmm interesting. I haven't previously seen a divergence like that.

For me if I reintroduce the braces, Android Studio does flag it. Here's a screenshot (in an unnaturally small window to make the screenshot readable:
![Screenshot from 2023-09-07 10-41-58](https://github.com/zulip/zulip-flutter/assets/28173/1dbd0e09-aa37-4696-a1b6-c3f6eea84653)

So it's there in the "Dart Analysis" pane at the bottom, and also in the list of issues at the top right of the editor pane (with the gray "!"-in-triangle icon — gray because it's an info-level issue). And the F2 key ("Next Highlighted Error") navigates to it.

Perhaps there's some Android Studio setting that affects that — like maybe that controls whether info-level issues are included? I don't remember changing such a setting, but it's possible I did long ago. If showing them isn't the default, it'd be good to put that setting in our `.idea/` directory if we can, so that everyone working in our tree gets it automatically.